x-pack/plugins/observability_solution/apm/public/components/app/service_tab_content/index.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { ReactNode } from 'react-markdown';
+import { EuiSpacer } from '@elastic/eui';
+import { SignalTypes } from '../../../../common/entities/types';
+import { useApmServiceContext } from '../../../context/apm_service/use_apm_service_context';
+import { isApmSignal, isLogsSignal } from '../../../utils/get_signal_type';
+import { useLocalStorage } from '../../../hooks/use_local_storage';
+import { emptyStateContent } from './constants';
+import { Tab } from './types';
+import { ServiceTabEmptyState } from '../service_tab_empty_state';
+import { SearchBar } from '../../shared/search_bar/search_bar';
+
+export interface ServiceTabContentProps {
+  tabName: Tab;
+  children: ReactNode;
+}
+
+export function ServiceTabContent({ children, tabName }: ServiceTabContentProps) {
+  const { serviceEntitySummary } = useApmServiceContext();
+
+  const hasOnlyLogsSignal =
+    serviceEntitySummary?.signalTypes &&
+    !isApmSignal(serviceEntitySummary.signalTypes as SignalTypes[]) &&
+    isLogsSignal(serviceEntitySummary.signalTypes as SignalTypes[]);
+
+  const [dismissedLogsOnlyEmptyState, setdismissedLogsOnlyEmptyState] = useLocalStorage(
+    `apm.dismissedLogsOnlyEmptyState.${tabName}`,
+    false
+  );
+
+  const displayEmptyStateOnly = tabName !== 'overview';
```

Comment:
nit: This works now (just for one tab) but this is not super flexible. 

since you're defining `emptyStateContent` per tab. What do you think of adding one more flag. ex `hideChildren` or `showOnlyEmptyState`


```
 overview: {
    title: i18n.translate('xpack.apm.serviceTabContent.serviceTabEmptyState.overviewTitle', {
      defaultMessage: 'Detect and resolve issues faster with deep visibility into your application',
    }),
    content: i18n.translate('xpack.apm.serviceTabContent.serviceTabEmptyState.overviewContent', {
      defaultMessage:
        'Understanding your application performance, relationships and dependencies by instrumenting with APM.',
    }),
    showOnlyEmptyState: false,
  },
```

and render the content according to this flag

Reply:
You’re absolutely right that it’s not scalable in this case. I hadn’t considered that this might be used later on since there weren’t supposed to be more changes initially.

I’ll go ahead and implement your suggestion, as it’s an improvement in itself.